src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -339,9 +339,9 @@ public void init() throws GitException, InterruptedException {
     @Override
     public boolean hasGitRepo() throws GitException, InterruptedException {
         if (hasGitRepo(".git")) {
-            // Check if this is a valid git repo with --is-inside-work-tree
+            // Check if this is a valid git repo with --resolve-git-dir
             try {
-                launchCommand("rev-parse", "--is-inside-work-tree");
+                launchCommand("rev-parse", "--resolve-git-dir",workspace.getAbsolutePath()+File.separator+".git");
```

Comment:
A bit of whitespace would be good for readability and line wrappings.

Just in case (sanity check) - can we expect the new CLI option to be available everywhere? (Is it not a recent addition to Git toolkit?) Otherwise maybe should handle a failure of such lookup and retry with one that nobody complained about ;)

Reply:
As per Git 1.8.0 documentation pages the --resolve-git-dir is supported, link I am referencing from https://code.google.com/archive/p/git-core/downloads?page=9

Reply:
Implemented the suggested changes.